packages/cli/src/deploy/resolveConfig.ts

```diff
@@ -44,7 +44,7 @@ export function resolveConfig<config extends ConfigInput>({
     .map(toFunctionSignature);
 
   const systems = Object.entries(resolvedConfig.systems).map(([systemName, system]): System => {
-    const namespace = config.namespace;
+    const namespace = system.namespace;
```

Comment:
does the deployer properly consider system namespaces when preparing to deploy this? registering namespace, etc

Reply:
Yes - but good point. In the deployer we [get every system ID](https://github.com/latticexyz/mud/blob/main/packages/cli/src/deploy/deploy.ts#L93) in the project then [register their namespace](https://github.com/latticexyz/mud/blob/main/packages/cli/src/deploy/ensureNamespaceOwner.ts#L17) if needed.